docs: document stickiness (#1124)
Unleash/unleash: website/docs/advanced/stickiness.md

---
id: stickiness
title: Stickiness
---

_Stickiness_ is how Unleash guarantees that the same user gets the same features every time. Stickiness is useful in any scenario where you want to either show a feature to only a subset of users or give users a variant of a feature.

## Calculation

> By default, Unleash calculates stickiness based on the user id and the group id. If the user id is unavailable, it falls back to using the session id instead. It hashes these values to a number between 0 and 100. This number is what determines whether a user will see a specific feature or variant. Because the process is deterministic, the same user will always get the same number.

Reviewer: Do we want to point out that we use the murmur hash, or is that not interesting here?

Author: I don't know, actually, but I like the suggestion. I'll add it in! 🙋🏾

If both the user id and the session id is unavailable, the calculation returns a random value and stickiness is not guaranteed.

## Consistency

Because the number assigned to a user won't change, Unleash also guarantees that the a user will keep seeing the same features even if certain other parameters change.

For instance: When using the [gradual rollout activation strategy](../user_guide/activation-strategies.md#gradual-rollout), any user whose number is less than or equal to the rollout percentage will see the feature. This means that the same users will keep seeing the feature even as you increase the percentage of your user base that sees the feature.

## Custom stickiness (beta)

:::info

> Custom stickiness is available Unleash Enterprise v4 and later.

Reviewer: ```suggestion
Custom stickiness is available starting from Unleash Enterprise v4.
```
